dvc/commands/plots.py

```diff
@@ -341,6 +349,12 @@ def _add_props_arguments(parser):
     parser.add_argument(
         "--y-label", default=None, help="Y axis label", metavar="<text>"
     )
+    parser.add_argument(
+        "--from-config",
+        help="Plot configuration file",
+        default=None,
+        metavar="<path>",
+    )
```

Comment:
Is there a need for this at the moment? 

If not, I'd suggest we drop it because the behavior isn't that clear to me. For example, if I do `dvc plots show plots/roc.json --from-config plots.yaml`, what if there are multiple plots that use `plots/roc.json`? If I do `dvc plots diff HEAD --from-config plots.yaml`, does HEAD get configured with `plots.yaml` from the workspace or from HEAD?

Reply:
There is no need, but also there has been some reluctance in introducing `plots` into `dvc.yaml` during our discussions. I would keep that and discuss this among the whole team. We can always deprecate `from-config`. Reverting changes for `dvc.yaml` will not be that easy. I was actually thinking about providing `--from-config` as initial workflow interface, like a playground, to "design" the plots. Once the plots are working as intended, the user could copy-paste them into the `dvc.yaml`

Reply:
Don't want to go in circles about the "dvc.yaml section" vs "external file" discussion but, given how tricky the error handling can get (i.e. https://github.com/iterative/dvc/issues/6150), I would prefer to be able to call `dvc plots --from-config` without getting errors from unrelated pipeline issues.

Reply:
> Reverting changes for `dvc.yaml` will not be that easy.

I didn't get what you mean by this. Could you explain?

> Don't want to go in circles about the "dvc.yaml section" vs "external file" discussion but, given how tricky the error handling can get (i.e. #6150), I would prefer to be able to call `dvc plots --from-config` without getting errors from unrelated pipeline issues

Agreed, but it seems out of scope and it doesn't work right now:

```
$ echo 'roc.json:\n  x: val_acc' > roc.json
$ echo '{"train": [{"val_acc": 0}]}' > roc.json
$ echo 'roc.json:\n  x: val_acc' > plots.json
$ dvc plots show --from-config plots.json
file:///Users/dave/repo/dvc_plots/index.html
$ echo invalid > dvc.yaml
$ dvc plots show --from-config plots.json
DVC failed to load some plots for following revisions: 'workspace'.
file:///Users/dave/repo/dvc_plots/index.html
```

The reason I want to exclude it (or hide it) is that it's a separate scenario that requires more testing and time, and I don't want to delay merging flexible plots to get `--from-config` right.

Reply:
> I didn't get what you mean by this. Could you explain?

Once we allow defining plots in `dvc.yaml`, practically there is no coming back from it before a major release (even if we deprecate it, we still need to allow for it for users that already adopted it). On the other hand I don't have a viable alternative - we don't have a way to make `dvc plots show` use `from-config` files because they are not marked in any way as files containing plots definitions.

We could also think about supporting something like:

```
# dvc.yaml
plots:
   - config_file.yaml
``` 
which would load config_file.yaml. This allows for delegating this responsibility to particular files not related to the pipeline.

Reply:
Based on conversation with @pared, we will make `--from-config` hidden for now to not delay implementation.